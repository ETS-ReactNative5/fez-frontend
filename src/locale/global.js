import {APP_URL} from 'config';

/*

NOTE:
- text can be either plain text, eg text: 'Some text to display' or
- text can be formatted HTML text, eg text: (<div>Click here to search google: <a href='google.com'>search google</a></div>)
IMPORTANT: if currently text contains placeholders, eg any characters in square brackets, eg [noOfResults] it cannot be formatted with HTML tagsâ€™

- help objects have the following shape:
help: {
    title: 'About these metrics',
    text: (<div></div>),
    buttonLabel: 'OK'
}
- text can be plain or formatted HTML component with links/tags/etc
- if help is not required, delete help: {} fully (including closing '},')

*/

export default {
    global: {
        title: 'UQ eSpace DEMO',
        logo: 'https://static.uq.net.au/v1/logos/corporate/uq-logo-white.svg',
        loading: 'Loading...',
        loadingUserAccount: 'Loading user account...',
        mainNavButton: {
            tooltip: 'Main navigation',
            closeMenuLabel: 'Close menu'
        },
        authentication: {
            signInText: 'Log in',
            signOutText: 'Log out'
        },
        skipNav: {
            title: 'Skip Navigation',
            ariaLabel: 'Click to skip, or tab to progress to the navigation',
        },
        loginAlert: {
            title: 'You are not logged in',
            message: 'Login to UQ eSpace for full search results and more services.',
            type: 'info_outline',
            actionButtonLabel: 'Click to login'
        },
        genericErrorMessage: (message) => (
            `Error has occurred during request and request cannot be processed. ${message} Please contact eSpace administrators or try again later.`
        ),
        notRegisteredAuthorAlert: {
            title: 'You are not registered in UQ eSpace as an author',
            message: 'Please contact the UQ eSpace administrator to resolve this.',
            type: 'info_outline'
        },
        noOrcidAlert: {
            title: 'ORCID ID REQUIRED',
            message: 'You do not have an ORCID ID linked to your UQ eSpace',
            type: 'info_outline',
            actionButtonLabel: 'Click here to link or register'
        },
        discardFormChangesConfirmation: {
            confirmationTitle: 'Discard changes?',
            confirmationMessage: 'Are you sure you want to navigate away and discard all changes to the form?',
            cancelButtonLabel: 'No',
            confirmButtonLabel: 'Yes'
        },
        linkWillOpenInNewWindow: 'Link to [destination] will open in a new window.',
        sources: {
            espace: {
                id: 'espace',
                title: 'eSpace',
                priority: 0,
                externalUrl: APP_URL + 'view/[id]',
                idKey: 'rek_pid'
            },
            wos: {
                id: 'wos',
                title: 'Web of science',
                priority: 1,
                externalUrl: 'http://ezproxy.library.uq.edu.au/login?url=http://gateway.isiknowledge.com/gateway/Gateway.cgi?GWVersion=2&SrcApp=resolve1&DestLinkType=FullRecord&DestApp=WOS_CPL&KeyUT=[id]&SrcAuth=uqueensland',
                idKey: 'fez_record_search_key_isi_loc.rek_isi_loc'
            },
            scopus: {
                id: 'scopus',
                title: 'Scopus',
                priority: 2,
                externalUrl: 'http://ezproxy.library.uq.edu.au/login?url=http://www.scopus.com/record/display.url?eid=[id]&origin=inward',
                idKey: 'fez_record_search_key_scopus_id.rek_scopus_id'
            },
            pubmed: {
                id: 'pubmed',
                title: 'PubMed',
                priority: 3,
                externalUrl: 'http://ezproxy.library.uq.edu.au/login?url=https://www.ncbi.nlm.nih.gov/pubmed/[id]',
                idKey: 'fez_record_search_key_pubmed_id.rek_pubmed_id'
            },
            crossref: {
                id: 'crossref',
                title: 'Crossref',
                priority: 4,
                externalUrl: 'http://ezproxy.library.uq.edu.au/login?url=https://doi.org/[id]',
                idKey: 'fez_record_search_key_doi.rek_doi'
            },
        },
        doiCitationLink: {
            ariaLabel: 'Open this DOI in an new window',
            prefix: 'https://doi.org/',
            externalUrl: 'https://doi.org/[id]',
        },
        pubmedCentralLink: {
            ariaLabel: 'Full text via Pubmed Central (open access)',
            prefix: 'https://www.ncbi.nlm.nih.gov/pmc/articles/',
            externalUrl: 'https://www.ncbi.nlm.nih.gov/pmc/articles/[id]'
        },
        embargoDateFormat: 'YYYY-MM-DD',
        defaultLinkDescription: 'Link to publication'
    }
};
