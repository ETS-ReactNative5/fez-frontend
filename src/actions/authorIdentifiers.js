import * as actions from './actionTypes';
import {get, patch} from 'repositories/generic';
import * as routes from 'repositories/routes';

import {transformAuthorIdentifier} from './authorIdentifierTransformer';
import {APP_URL, AUTHOR_IDENTIFIER_ORCID} from 'config/general';
import {locale} from 'locale';
import {dismissAppAlert} from './app';

/**
 * Returns orcid access token for an author and updates author details with linked ORCID id
 * @param {string} userId
 * @param {string} authorId
 * @param {string} orcidCode - code return in query string from ORCID authorisation process
 * @returns {action}
 */
export function linkAuthorOrcidId(userId, authorId, orcidCode) {
    return dispatch => {
        dispatch({type: actions.CURRENT_AUTHOR_SAVING});

        dispatch({
            type: actions.APP_ALERT_SHOW,
            payload: {
                ...locale.pages.orcidLink.progressAlert
            }
        });

        let orcidId = null;

        // parameters required for AUTHOR_ORCID_DETAILS_API call
        const params = {
            code: orcidCode,
            redirUri: APP_URL
        };

        // get ORCID id for current user
        return get(routes.AUTHOR_ORCID_DETAILS_API({userId: userId, params: params}), false)
            .then((response) => {
                orcidId = response.data.orcid;
                if (!orcidId) return Promise.reject({message: 'ORCID id is missing from response. '});
                const authorPatchRequest = transformAuthorIdentifier(AUTHOR_IDENTIFIER_ORCID, authorId, orcidId, response.data);
                // patch author record with corresponding ORCID id
                return patch(routes.AUTHOR_API({authorId}), authorPatchRequest);
            })
            .then((response) => {
                // author details saved successfully
                dispatch({
                    type: actions.CURRENT_AUTHOR_SAVED,
                    payload: response.data
                });

                dispatch({
                    type: actions.APP_ALERT_SHOW,
                    payload: {
                        ...locale.pages.orcidLink.successAlert,
                        dismissAction: () => dispatch(dismissAppAlert())
                    }
                });
            })
            .catch(error => {
                console.log(error);
                if (error.status === 403) dispatch({type: actions.CURRENT_ACCOUNT_ANONYMOUS});

                dispatch({
                    type: actions.CURRENT_AUTHOR_SAVE_FAILED,
                    payload: error.message
                });

                dispatch({
                    type: actions.APP_ALERT_SHOW,
                    payload: {
                        ...{...locale.pages.orcidLink, message: locale.pages.orcidLink.message ? locale.pages.orcidLink.message(error.message) : error.message},
                        dismissAction: () => dispatch(dismissAppAlert())
                    }
                });
            });
    };
}

