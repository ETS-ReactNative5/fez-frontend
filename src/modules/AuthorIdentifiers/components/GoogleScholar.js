import React from 'react';
import PropTypes from 'prop-types';
import {propTypes} from 'redux-form/immutable';
import {Field} from 'redux-form/immutable';
import RaisedButton from 'material-ui/RaisedButton';
import {TextField} from 'uqlibrary-react-toolbox/build/TextField';
import {StandardCard} from 'uqlibrary-react-toolbox/build/StandardCard';
import {StandardPage} from 'uqlibrary-react-toolbox/build/StandardPage';
import {Alert} from 'uqlibrary-react-toolbox/build/Alert';

import {locale} from 'locale';
import {routes, validation} from 'config';

export default class GoogleScholar extends React.PureComponent {
    static propTypes = {
        ...propTypes,
        author: PropTypes.object,
        accountAuthorLoading: PropTypes.bool,
        history: PropTypes.object.isRequired,
        actions: PropTypes.object.isRequired
    };

    componentWillReceiveProps(nextProps) {
        if (nextProps.submitSucceeded !== this.props.submitSucceeded) {
            // show app level alert on success
            this.props.actions.showAppAlert({
                ...locale.pages.googleScholarLink.successAlert,
                dismissAction: this.props.actions.dismissAppAlert
            });
            this._navigateToDashboard();
        }
    }

    _handleKeyboardFormSubmit = (event) => {
        if (event.key === 'Enter' && !event.shiftKey) {
            event.preventDefault();
            this.props.handleSubmit();
        }
    };

    _navigateToDashboard = () => {
        this.props.history.push(routes.pathConfig.dashboard);
    };

    getAlert = ({submitFailed = false, submitting = false, error, submitSucceeded = false, alertLocale = {}}) => {
        let alertProps = null;
        if (submitFailed && error) {
            alertProps = {
                ...alertLocale.errorAlert,
                message: alertLocale.errorAlert.message
                    ? alertLocale.errorAlert.message(error)
                    : error
            };
        } else if (submitting) {
            alertProps = {...alertLocale.progressAlert};
        } else if (submitSucceeded) {
            alertProps = {...alertLocale.successAlert};
        }
        return alertProps ? (<Alert {...alertProps} />) : null;
    };

    render() {
        // wait for account details to be loaded
        if(this.props.accountAuthorLoading || !this.props.author) {
            return (<div />);
        }

        const txt = locale.pages.googleScholarLink;
        const cardLocale = !this.props.author.aut_google_scholar_id
            ? txt.add
            : txt.edit;

        return (
            <StandardPage title={txt.title}>
                <form onKeyDown={this._handleKeyboardFormSubmit}>
                    <StandardCard title={cardLocale.title} help={txt.help}>
                        {cardLocale.description}
                        <div className="columns">
                            <div className="column">
                                <Field
                                    component={TextField}
                                    disabled={this.props.submitting}
                                    name="aut_google_scholar_id"
                                    fullWidth
                                    {...txt.labels.googleScholarIdField}
                                    validate={[validation.required, validation.isValidGoogleScholarId]}
                                    className="requiredField" />
                            </div>
                        </div>
                    </StandardCard>

                    {
                        this.getAlert({...this.props, alertLocale: txt})
                    }

                    <div className="columns action-buttons">
                        <div className="column is-hidden-mobile"/>
                        <div className="column is-narrow-desktop">
                            <RaisedButton
                                fullWidth
                                disabled={this.props.submitting}
                                label={txt.labels.cancel}
                                onTouchTap={this._navigateToDashboard} />
                        </div>
                        <div className="column is-narrow-desktop">
                            <RaisedButton
                                secondary
                                fullWidth
                                disabled={this.props.submitting || this.props.invalid}
                                label={txt.labels.submit}
                                onTouchTap={this.props.handleSubmit} />
                        </div>
                    </div>
                </form>
            </StandardPage>
        );
    }
}
