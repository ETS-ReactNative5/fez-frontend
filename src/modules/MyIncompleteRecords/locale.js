import React from 'react';
import locale from 'locale/components';

/*

NOTE:
- text can be either plain text, eg text: 'Some text to display' or
- text can be formatted HTML text, eg
    text: (<div>Click here to search google: <a href='google.com'>search google</a></div>)
IMPORTANT: if currently text contains placeholders, eg any characters in square brackets,
eg [noOfResults] it cannot be formatted with HTML tags’

- help objects have the following shape:
help: {
    title: 'About these metrics',
    text: (<div></div>),
    buttonLabel: 'CLOSE'
}
- text can be plain or formatted HTML component with links/tags/etc
- if help is not required, delete help: {} fully (including closing '},')

*/
/* eslint-disable max-len */
export default {
    title: 'Complete my work',
    help: {
        title: 'Add more information to an existing NTRO',
        text: (
            <div>
                For help, click{' '}
                <a
                    href="https://guides.library.uq.edu.au/for-researchers/uqespace-publications-datasets/add-missing-ntro-information"
                    target="_blank"
                >
                    here
                </a>
                .
            </div>
        ),
        buttonLabel: 'CLOSE',
    },
    submitButtonLabel: 'Complete work',
    cancelButtonLabel: 'Cancel and return to my incomplete works list',
    fields: {
        notes: {
            title: 'Notes',
            label: 'Notes for this work',
            placeholder: 'Add any other notes or comments about this work to send to the eSpace team.',
        },
        grants: {
            title: 'Grant information',
        },
        authors: {
            ...locale.components.authors,
            description: (
                <span>
                    For each author marked with a <span style={{ color: '#e60000', weight: 'bold' }}>red</span> prompt,
                    select the author name at Step 1, add the affiliation information as at time of publication at Step
                    2, then click <b>UPDATE AUTHOR</b>.
                </span>
            ),
            field: {
                ...locale.components.authors.field,
                form: {
                    locale: {
                        ...locale.components.authors.field.form.locale,
                        descriptionStep1: (
                            <div>
                                <span className="authorSteps" key="step-2">
                                    Step 2 of 2
                                </span>{' '}
                                - <b>Update the affiliation information</b>.
                            </div>
                        ),
                        addButton: 'Update author',
                    },
                },
                header: {
                    ...locale.components.authors.field.header,
                    locale: {
                        ...locale.components.authors.field.header.locale,
                        descriptionStep2: (
                            <div>
                                <span className="authorSteps" key="step-1">
                                    Step 1 of 2
                                </span>{' '}
                                - <b>Select an author</b>.
                            </div>
                        ),
                    },
                },
                row: {
                    ...locale.components.authors.field.row,
                    locale: {
                        ...locale.components.authors.field.row.locale,
                        selectHint: 'Select this author ([name]) to update their affiliation data.',
                        requiredLabel: "This author's data is incomplete.",
                    },
                },
            },
        },
        fileUpload: {
            title: 'Upload files',
            locale: {
                instructions: '',
                accessTermsAndConditions:
                    'I understand that the files indicated above as open access will be submitted as open access and will be made publicly available immediately or will be made available on the indicated embargo date.  All other files submitted will be accessible by UQ eSpace administrators.',
                validation: {
                    ['notFiles']: 'Invalid files ([fileNames])',
                    ['invalidFileNames']: 'File(s) ([fileNames]) have invalid file name',
                    ['tooBigFiles']: 'File(s) ([fileNames]) exceed maximum allowed upload file size',
                    ['tooManyFiles']:
                        'Maximum number of files ([maxNumberOfFiles]) has been exceeded. File(s) ([fileNames]) will not be uploaded',
                    ['duplicateFiles']: 'File(s) ([fileNames]) are duplicates and have been ignored',
                },
                successTitle: 'Success',
                successMessage: 'Successfully added [numberOfFiles] file(s) to upload queue.',
                delayNotice: 'Notice',
                delayMessage:
                    'During peak times, there may be a delay before newly uploaded files appear on the record.',
                errorTitle: 'Upload Errors',
                fileUploadRestrictionHeading: 'File upload restrictions',
                fileUploadRestrictions: (
                    <div>
                        Please ensure:
                        <ul>
                            <li>file names begin with a letter and are less than 45 characters long</li>
                            <li>file names contain only upper and lowercase alphanumeric characters, and underscores</li>
                            <li>
                                file names have only a single period which precedes the file extension: e.g. “.pdf”, “.mov”,
                                “.tiff”, “.wav” etc.<br />
                                (two periods are allowed for multipart zip files. Allowed formats for part description are: e.g.
                                “.001 ... .999”, “.r01 ... .r999” and “.part1 ... .part999”)
                            </li>
                            <li>files are uploaded individually and not inside a folder</li>
                            <li>files are smaller than 8GB.</li>
                        </ul>
                    </div>
                ),
                fileUploadInstruction: <p>Click here to select files, or drag files into this area to upload</p>,
            },
            text: (
                <div>
                    <span className="requiredField">
                        <label>&nbsp;</label>
                    </span>
                </div>
            ),
        },
    },
    successWorkflowConfirmation: {
        confirmationTitle: 'Your work has been updated',
        fileFailConfirmationAlert: {
            title: 'UPLOAD FAILED',
            message: 'File upload and/or notes post failed',
            type: 'warning',
        },
        cancelButtonLabel: 'Complete another work',
        confirmButtonLabel: 'Go to my dashboard',
    },
    prompt: {
        title: 'Missing data',
        message:
            'follow the red prompts to add missing information. You can also provide additional grant details or include notes to the eSpace team.',
        type: 'info_outline',
    },
    progressAlert: {
        type: 'info_outline',
        title: 'Saving',
        message: 'Updating work is in progress.',
        showLoader: true,
    },
    successAlert: {
        type: 'done',
        title: 'Success',
        message: 'New information has been saved successfully.',
    },
};
