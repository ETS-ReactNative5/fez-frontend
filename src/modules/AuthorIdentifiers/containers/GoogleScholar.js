import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {reduxForm, getFormValues, SubmissionError} from 'redux-form/immutable';
import Immutable from 'immutable';
import {withRouter} from 'react-router-dom';
import * as actions from 'actions';

import GoogleScholar from '../components/GoogleScholar';
const FORM_NAME = 'GoogleScholar';

const onSubmit = (values, dispatch, props) => {
    return dispatch(actions.updateCurrentAuthor(props.author.aut_id, values.toJS()))
        .then(() => {
            // once this promise is resolved form is submitted successfully and will call parent container
            // reported bug to redux-form:
            // reset form after success action was dispatched:
            // componentWillUnmount cleans up form, but then onSubmit success sets it back to active
            // setTimeout(()=>{
            //     dispatch(reset(FORM_NAME));
            // }, 100);
        }).catch(error => {
            throw new SubmissionError({_error: error.message});
        });
};

const mapStateToProps = (state) => (
    {
        ...state.get('accountReducer'),
        formValues: getFormValues(FORM_NAME)(state) || Immutable.Map({}),
        initialValues: {
            aut_id: state.get('accountReducer') && state.get('accountReducer').author
                ? state.get('accountReducer').author.aut_id
                : '',
            aut_google_scholar_id: state.get('accountReducer') && state.get('accountReducer').author
                ? state.get('accountReducer').author.aut_google_scholar_id
                : ''
        }
    }
);

const mapDispatchToProps = (dispatch) => (
    {
        actions: bindActionCreators(actions, dispatch)
    }
);

let GoogleScholarContainer = reduxForm({
    form: FORM_NAME,
    onSubmit
})(GoogleScholar);

GoogleScholarContainer = connect(mapStateToProps, mapDispatchToProps)(GoogleScholarContainer);
GoogleScholarContainer = withRouter(GoogleScholarContainer);

export default GoogleScholarContainer;
