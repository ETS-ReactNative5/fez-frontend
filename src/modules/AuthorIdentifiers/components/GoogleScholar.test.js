import React from 'react';

import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';
import injectTapEventPlugin from 'react-tap-event-plugin';
import Immutable from 'immutable';
import {MemoryRouter} from 'react-router-dom'
import {Provider} from 'react-redux';

import GoogleScholar from './GoogleScholar';
import {testAuthor} from 'mock/data/testing/authorIdentifiers';
import {locale} from 'locale';

const create = () => {
    const initialState = Immutable.Map();

    const store = {
        getState: jest.fn(() => (initialState)),
        dispatch: jest.fn(),
        subscribe: jest.fn()
    };
    const next = jest.fn();
    const invoke = (action) => thunk(store)(next)(action);
    return {store, next, invoke}
};

function setup(testProps, isShallow = true) {
    const props = {
        ...testProps,

        accountAuthorLoading: testProps.accountAuthorLoading || false,
        author: testProps.author || testAuthor,
        actions: testProps.actions || {showAppAlert: jest.fn()},
        history: testProps.history || {push: jest.fn()},

        // redux form props
        handleSubmit: testProps.handleSubmit || jest.fn(),
        initialValues: testProps.initialValues || {
            aut_id: !!testProps.author ? testProps.author.aut_id : testAuthor.aut_id,
            aut_google_scholar_id: !!testProps.author ? testProps.author.aut_google_scholar_id : testAuthor.aut_google_scholar_id
        },
        submitSucceeded: testProps.submitSucceeded || false,
        submitFailed: testProps.submitFailed || false,
        error: testProps.error || null
    };

    if(isShallow) {
        return shallow(<GoogleScholar {...props} />);
    }

    return mount(
        <Provider store={create().store}>
            <MemoryRouter>
                <GoogleScholar {...props} />
            </MemoryRouter>
        </Provider>, {
            context: {
                muiTheme: getMuiTheme(),

            },
            childContextTypes: {
                muiTheme: PropTypes.object.isRequired
            }
        });
}

beforeAll(() => {
    injectTapEventPlugin();
});

describe('Component GoogleScholar ', () => {

    it('should render nothing if author is not loaded', () => {
        const wrapper = setup({accountAuthorLoading: true, author: null});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render form if author doesn\'t have google scholar id', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('.requiredField').length).toEqual(1);
    });

    it('should render form if author has google scholar id', () => {
        const wrapper = setup({author: { ...testAuthor, aut_google_scholar_id: 'abc123'}});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('.requiredField').length).toEqual(1);
    });

    it('should redirect to the dashbaord', () => {
        const testMethod = jest.fn();
        const wrapper = setup({history: {push: testMethod}});
        wrapper.instance()._navigateToDashboard();
        expect(testMethod).toHaveBeenCalledWith('/dashboard');
    });

    it('should submit form when user hits Enter', () => {
        const testMethod = jest.fn();
        const wrapper = setup({handleSubmit: testMethod});
        wrapper.instance()._handleKeyboardFormSubmit({key: 'Enter', preventDefault: jest.fn()});
        expect(testMethod).toHaveBeenCalled();
    });

    it('should NOT submit the form when user hits shift+Enter', () => {
        const testMethod = jest.fn();
        const wrapper = setup({handleSubmit: testMethod});
        wrapper.instance()._handleKeyboardFormSubmit({key: 'Enter', shiftKey: true, preventDefault: jest.fn()});
        expect(testMethod).not.toHaveBeenCalled();
    });

    it('should go back to the dashboard if the submission succeeded', () => {
        const testMethod = jest.fn();
        const wrapper = setup({history: {push: testMethod}});
        wrapper.setProps({submitSucceeded: true});
        expect(testMethod).toHaveBeenCalledWith('/dashboard');
    });

    it('should dispatch action to display success alert', () => {
        const testMethod = jest.fn();
        const wrapper = setup({actions: {showAppAlert: testMethod}});
        wrapper.setProps({submitSucceeded: true});
        expect(testMethod).toHaveBeenCalled();
    });

    it('should display submission error if saving failed', () => {
        const testMethod = jest.fn();
        const wrapper = setup({submitFailed: true, error: 'failed!'});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should display submission in progress alert', () => {
        const testMethod = jest.fn();
        const wrapper = setup({submitting: true});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});
