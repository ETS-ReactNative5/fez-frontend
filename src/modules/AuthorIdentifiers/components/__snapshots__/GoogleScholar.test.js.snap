// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component GoogleScholar  should display submission error if saving failed 1`] = `
<StandardPage
  title="Google Scholar identifier"
>
  <form
    onKeyDown={[Function]}
  >
    <StandardCard
      className=""
      help={
        Object {
          "buttonLabel": "OK",
          "text": <div>
            <h3>
              How to create the profile?
            </h3>
            <ul>
              <li>
                Sign to your Google account, or create one if you don't have one.
              </li>
              <li>
                After you sign in, the Citations sign up form will ask you to confirm the spelling of your name, to enter your affiliation, etc.
              </li>
              <li>
                On the next page, you will see a list of articles. Add the articles that are yours.
              </li>
              <li>
                Once you're done with adding articles, it will ask you what to do when the article data changes in Google Scholar. You can either have the updates applied to your profile automatically or you can choose to review them beforehand.
              </li>
              <li>
                Finally, you will see your profile.Once you are satisfied with the results, make your profile public.
              </li>
            </ul>
          </div>,
          "title": "Google identifier help",
        }
      }
      title="Add your Google Scholar identifier"
    >
      <div>
        <p>
          Creating your Google Scholar Citation profile will make sure that Google Scholar will easily and accurately group all the citations of your publications into one pool. A profile generally lists your name, chosen keywords of research interest, generated citation metrics, and citations (including links to citing articles).
        </p>
        <p>
          In order to create a Google Scholar Citation profile, you need a Google Account.
        </p>
        <p>
          For more information see the 
          <a
            href="http://guides.library.uq.edu.au/researcher-identifiers/google-scholar-citations-profile"
            target="_blank"
          >
            Google Scholar Citations help page.
          </a>
        </p>
      </div>
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <Field
            className="requiredField"
            component={[Function]}
            floatingLabelText="Google Scholar ID"
            fullWidth={true}
            hintText="Enter your Google Scholar ID"
            name="aut_google_scholar_id"
            validate={
              Array [
                [Function],
                [Function],
              ]
            }
          />
        </div>
      </div>
    </StandardCard>
    <Alert
      allowDismiss={false}
      message="Error has occurred during request and request cannot be processed. failed! Please contact eSpace administrators or try again later."
      title="Error"
      type="error_outline"
    />
    <div
      className="columns action-buttons"
    >
      <div
        className="column is-hidden-mobile"
      />
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={false}
          fullWidth={true}
          label="Cancel"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={false}
        />
      </div>
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={false}
          fullWidth={true}
          label="Save Google Scholar ID"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={true}
        />
      </div>
    </div>
  </form>
</StandardPage>
`;

exports[`Component GoogleScholar  should display submission in progress alert 1`] = `
<StandardPage
  title="Google Scholar identifier"
>
  <form
    onKeyDown={[Function]}
  >
    <StandardCard
      className=""
      help={
        Object {
          "buttonLabel": "OK",
          "text": <div>
            <h3>
              How to create the profile?
            </h3>
            <ul>
              <li>
                Sign to your Google account, or create one if you don't have one.
              </li>
              <li>
                After you sign in, the Citations sign up form will ask you to confirm the spelling of your name, to enter your affiliation, etc.
              </li>
              <li>
                On the next page, you will see a list of articles. Add the articles that are yours.
              </li>
              <li>
                Once you're done with adding articles, it will ask you what to do when the article data changes in Google Scholar. You can either have the updates applied to your profile automatically or you can choose to review them beforehand.
              </li>
              <li>
                Finally, you will see your profile.Once you are satisfied with the results, make your profile public.
              </li>
            </ul>
          </div>,
          "title": "Google identifier help",
        }
      }
      title="Add your Google Scholar identifier"
    >
      <div>
        <p>
          Creating your Google Scholar Citation profile will make sure that Google Scholar will easily and accurately group all the citations of your publications into one pool. A profile generally lists your name, chosen keywords of research interest, generated citation metrics, and citations (including links to citing articles).
        </p>
        <p>
          In order to create a Google Scholar Citation profile, you need a Google Account.
        </p>
        <p>
          For more information see the 
          <a
            href="http://guides.library.uq.edu.au/researcher-identifiers/google-scholar-citations-profile"
            target="_blank"
          >
            Google Scholar Citations help page.
          </a>
        </p>
      </div>
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <Field
            className="requiredField"
            component={[Function]}
            disabled={true}
            floatingLabelText="Google Scholar ID"
            fullWidth={true}
            hintText="Enter your Google Scholar ID"
            name="aut_google_scholar_id"
            validate={
              Array [
                [Function],
                [Function],
              ]
            }
          />
        </div>
      </div>
    </StandardCard>
    <Alert
      allowDismiss={false}
      message="Request is being processed."
      title="Saving"
      type="info_outline"
    />
    <div
      className="columns action-buttons"
    >
      <div
        className="column is-hidden-mobile"
      />
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={true}
          fullWidth={true}
          label="Cancel"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={false}
        />
      </div>
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={true}
          fullWidth={true}
          label="Save Google Scholar ID"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={true}
        />
      </div>
    </div>
  </form>
</StandardPage>
`;

exports[`Component GoogleScholar  should render form if author doesn't have google scholar id 1`] = `
<StandardPage
  title="Google Scholar identifier"
>
  <form
    onKeyDown={[Function]}
  >
    <StandardCard
      className=""
      help={
        Object {
          "buttonLabel": "OK",
          "text": <div>
            <h3>
              How to create the profile?
            </h3>
            <ul>
              <li>
                Sign to your Google account, or create one if you don't have one.
              </li>
              <li>
                After you sign in, the Citations sign up form will ask you to confirm the spelling of your name, to enter your affiliation, etc.
              </li>
              <li>
                On the next page, you will see a list of articles. Add the articles that are yours.
              </li>
              <li>
                Once you're done with adding articles, it will ask you what to do when the article data changes in Google Scholar. You can either have the updates applied to your profile automatically or you can choose to review them beforehand.
              </li>
              <li>
                Finally, you will see your profile.Once you are satisfied with the results, make your profile public.
              </li>
            </ul>
          </div>,
          "title": "Google identifier help",
        }
      }
      title="Add your Google Scholar identifier"
    >
      <div>
        <p>
          Creating your Google Scholar Citation profile will make sure that Google Scholar will easily and accurately group all the citations of your publications into one pool. A profile generally lists your name, chosen keywords of research interest, generated citation metrics, and citations (including links to citing articles).
        </p>
        <p>
          In order to create a Google Scholar Citation profile, you need a Google Account.
        </p>
        <p>
          For more information see the 
          <a
            href="http://guides.library.uq.edu.au/researcher-identifiers/google-scholar-citations-profile"
            target="_blank"
          >
            Google Scholar Citations help page.
          </a>
        </p>
      </div>
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <Field
            className="requiredField"
            component={[Function]}
            floatingLabelText="Google Scholar ID"
            fullWidth={true}
            hintText="Enter your Google Scholar ID"
            name="aut_google_scholar_id"
            validate={
              Array [
                [Function],
                [Function],
              ]
            }
          />
        </div>
      </div>
    </StandardCard>
    <div
      className="columns action-buttons"
    >
      <div
        className="column is-hidden-mobile"
      />
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={false}
          fullWidth={true}
          label="Cancel"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={false}
        />
      </div>
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={false}
          fullWidth={true}
          label="Save Google Scholar ID"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={true}
        />
      </div>
    </div>
  </form>
</StandardPage>
`;

exports[`Component GoogleScholar  should render form if author has google scholar id 1`] = `
<StandardPage
  title="Google Scholar identifier"
>
  <form
    onKeyDown={[Function]}
  >
    <StandardCard
      className=""
      help={
        Object {
          "buttonLabel": "OK",
          "text": <div>
            <h3>
              How to create the profile?
            </h3>
            <ul>
              <li>
                Sign to your Google account, or create one if you don't have one.
              </li>
              <li>
                After you sign in, the Citations sign up form will ask you to confirm the spelling of your name, to enter your affiliation, etc.
              </li>
              <li>
                On the next page, you will see a list of articles. Add the articles that are yours.
              </li>
              <li>
                Once you're done with adding articles, it will ask you what to do when the article data changes in Google Scholar. You can either have the updates applied to your profile automatically or you can choose to review them beforehand.
              </li>
              <li>
                Finally, you will see your profile.Once you are satisfied with the results, make your profile public.
              </li>
            </ul>
          </div>,
          "title": "Google identifier help",
        }
      }
      title="Update your Google Scholar identifier"
    >
      <div>
        <p>
          Update your Google Scholar ID below.
        </p>
        <p>
          For more information see the 
          <a
            href="http://guides.library.uq.edu.au/researcher-identifiers/google-scholar-citations-profile"
            target="_blank"
          >
            Google Scholar Citations help page.
          </a>
        </p>
      </div>
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <Field
            className="requiredField"
            component={[Function]}
            floatingLabelText="Google Scholar ID"
            fullWidth={true}
            hintText="Enter your Google Scholar ID"
            name="aut_google_scholar_id"
            validate={
              Array [
                [Function],
                [Function],
              ]
            }
          />
        </div>
      </div>
    </StandardCard>
    <div
      className="columns action-buttons"
    >
      <div
        className="column is-hidden-mobile"
      />
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={false}
          fullWidth={true}
          label="Cancel"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={false}
        />
      </div>
      <div
        className="column is-narrow-desktop"
      >
        <RaisedButton
          disabled={false}
          fullWidth={true}
          label="Save Google Scholar ID"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={true}
        />
      </div>
    </div>
  </form>
</StandardPage>
`;

exports[`Component GoogleScholar  should render nothing if author is not loaded 1`] = `<div />`;
